src/auxinfo/auxinfo_commit.rs

```diff
@@ -32,43 +35,100 @@ impl AuxInfoCommit {
 
 #[derive(Serialize, Deserialize, Clone)]
 pub(crate) struct AuxInfoDecommit {
-    pub sid: Identifier,
-    pub sender: ParticipantIdentifier,
-    pub rid: [u8; 32],
-    pub u_i: [u8; 32],
-    pub pk: AuxInfoPublic,
+    sid: Identifier,
+    sender: ParticipantIdentifier,
+    rid: [u8; 32],
+    u_i: [u8; 32],
+    public_keys: AuxInfoPublic,
+}
+
+impl Debug for AuxInfoDecommit {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // Redacting rid and u_i because I'm not sure how sensitive they are. If later
+        // analysis suggests they're fine to print, please udpate accordingly.
+        f.debug_struct("AuxInfoDecommit")
+            .field("sid", &self.sid)
+            .field("sender", &self.sender)
+            .field("rid", &"[redacted]")
+            .field("u_i", &"[redacted]")
+            .field("public keys", &self.public_keys)
+            .finish()
+    }
 }
 
 impl AuxInfoDecommit {
     pub(crate) fn new<R: RngCore + CryptoRng>(
         rng: &mut R,
         sid: &Identifier,
         sender: &ParticipantIdentifier,
-        pk: &AuxInfoPublic,
-    ) -> Self {
+        public_keys: AuxInfoPublic,
+    ) -> Result<Self> {
         let mut rid = [0u8; 32];
         let mut u_i = [0u8; 32];
         rng.fill_bytes(rid.as_mut_slice());
         rng.fill_bytes(u_i.as_mut_slice());
-        Self {
+
+        public_keys.verify()?;
+        if sender != public_keys.participant() {
+            error!("Created AuxInfoDecommit with different participant IDs in the sender and public_keys fields");
+            return Err(InternalError::InternalInvariantFailed);
+        }
+
+        Ok(Self {
             sid: *sid,
             sender: *sender,
             rid,
             u_i,
-            pk: pk.clone(),
-        }
+            public_keys,
+        })
     }
 
     pub(crate) fn from_message(message: &Message) -> Result<Self> {
         if message.message_type() != MessageType::Auxinfo(AuxinfoMessageType::R2Decommit) {
             return Err(InternalError::MisroutedMessage);
         }
         let auxinfo_decommit: AuxInfoDecommit = deserialize!(&message.unverified_bytes)?;
+
+        // Public parameters in this decommit must be consistent with each other
+        auxinfo_decommit.public_keys.verify()?;
+
+        // Owner must be consistent across message, public keys, and decommit
+        if *auxinfo_decommit.public_keys.participant() != auxinfo_decommit.sender {
+            error!(
+                "Deserialized AuxInfoDecommit has different participant IDs in the sender ({}) and public_keys ({}) fields",
+                auxinfo_decommit.sender,
+                auxinfo_decommit.public_keys.participant(),
+            );
+            return Err(InternalError::Serialization);
```

Comment:
I don't know if this is necessarily a `Serialization` error. Instead, it should be an `InternalInvariantFailed` error? In my mind `Serialization` errors are only for when bytes fail to deserialize into some expected type.

Reply:
Gotcha. Yeah `ProtcolError` sounds right.